### ISSUE TITLE
[ML] Implement personalized model training using historical transaction data

### ISSUE THREAD

**Issue by wzh4464** [wzh4464/auto-expense-tracker#9]
**Description**: 
Build ML optimization pipeline to improve extraction accuracy using user's existing 2-month MOZE transaction history through fine-tuning and personalized prompt templates.

**Acceptance Criteria**:
- [ ] Parse MOZE export data to extract user patterns
- [ ] Generate personalized prompt templates with user examples
- [ ] Implement LoRA fine-tuning for local models
- [ ] A/B test optimized vs baseline extraction accuracy
- [ ] Achieve 10-15% accuracy improvement on user-specific patterns
- [ ] Support incremental learning from user corrections
- [ ] Create training data pipeline for future model updates

**Technical Implementation**:
```python
class PersonalizationEngine:
    def analyze_user_patterns(self, moze_export: str) -> UserPatterns
    def generate_personalized_prompts(self, patterns: UserPatterns) -> PromptTemplates
    def fine_tune_local_model(self, training_data: TrainingDataset) -> FineTunedModel
    def evaluate_improvement(self, baseline: Model, optimized: Model) -> Metrics
```

**Training Data Requirements**:
- Minimum 200+ confirmed transactions
- Merchant frequency analysis
- Category preference mapping
- User correction history integration

**Optimization Approaches**:
1. **Prompt Engineering**: Dynamic templates with user examples
2. **LoRA Fine-tuning**: Efficient adaptation of Phi-3/Qwen models
3. **Few-shot Learning**: Context injection with relevant historical examples

**Performance Targets**:
- 15% accuracy improvement on frequent merchants
- 20% better category prediction for user's spending patterns
- <5 second fine-tuning inference time
- Support for 500+ historical transactions

**Deliverables**:
- `src/ml/personalization_engine.py` module
- MOZE data parser and pattern analyzer
- LoRA training pipeline
- Personalized prompt template generator
- Evaluation metrics and A/B testing framework
